web/react/components/rhs_comment.jsx

```diff
@@ -0,0 +1,207 @@
+var client = require('../utils/client.jsx');
+var AsyncClient = require('../utils/async_client.jsx');
+var ActionTypes = Constants.ActionTypes;
+
+export default class RhsComment extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.retryComment = this.retryComment.bind(this);
+
+        this.state = {};
+    }
+    retryComment(e) {
+        e.preventDefault();
+
+        var post = this.props.post;
+        client.createPost(post, post.channel_id,
+            function success(data) {
+                AsyncClient.getPosts(post.channel_id);
+
+                var channel = ChannelStore.get(post.channel_id);
+                var member = ChannelStore.getMember(post.channel_id);
+                member.msg_count = channel.total_msg_count;
+                member.last_viewed_at = (new Date()).getTime();
+                ChannelStore.setChannelMember(member);
+
+                AppDispatcher.handleServerAction({
+                    type: ActionTypes.RECIEVED_POST,
+                    post: data
+                });
+            },
+            function fail() {
+                post.state = Constants.POST_FAILED;
+                PostStore.updatePendingPost(post);
+                this.forceUpdate();
+            }.bind(this)
+        );
+
+        post.state = Constants.POST_LOADING;
+        PostStore.updatePendingPost(post);
+        this.forceUpdate();
+    }
+    shouldComponentUpdate(nextProps) {
+        if (!utils.areStatesEqual(nextProps.post, this.props.post)) {
+            return true;
+        }
+
+        return false;
+    }
+    render() {
+        var post = this.props.post;
+
+        var currentUserCss = '';
+        if (UserStore.getCurrentId() === post.user_id) {
+            currentUserCss = 'current--user';
+        }
+
+        var isOwner = UserStore.getCurrentId() === post.user_id;
+
+        var type = 'Post';
+        if (post.root_id.length > 0) {
+            type = 'Comment';
+        }
+
+        var message = utils.textToJsx(post.message);
+        var timestamp = UserStore.getCurrentUser().update_at;
+
+        var loading;
+        var postClass = '';
+        if (post.state === Constants.POST_FAILED) {
+            postClass += ' post-fail';
+            loading = (
+                <a
+                    className='theme post-retry pull-right'
+                    href='#'
+                    onClick={this.retryComment}
+                >
+                    Retry
+                </a>
+            );
+        } else if (post.state === Constants.POST_LOADING) {
+            postClass += ' post-waiting';
+            loading = (
+                <img
+                    className='post-loading-gif pull-right'
+                    src='/static/images/load.gif'
+                />
+            );
+        }
+
+        var ownerOptions;
+        if (isOwner && post.state !== Constants.POST_FAILED && post.state !== Constants.POST_LOADING) {
```

Comment:
Lots of this is duplicate code from post_info.jsx. Probably should be refactored so we don't have duplicate code.